api/src/main/java/org/openmrs/Condition.java

```diff
@@ -0,0 +1,211 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs;
+
+import javax.persistence.AttributeOverride;
+import javax.persistence.AttributeOverrides;
+import javax.persistence.Column;
+import javax.persistence.Embedded;
+import java.util.Date;
+
+/**
+ * The condition class records detailed information about a condition, problem, diagnosis, or other situation or issue.
+ * This records information about a disease/illness identified from diagnosis
+ * or identification of health issues/situations that require ongoing monitoring.
+ * 
+ * @see <a href="https://www.hl7.org/fhir/condition.html">https://www.hl7.org/fhir/condition.html</a>
+ * 
+ * @since 2.2
+ */
+public class Condition extends BaseChangeableOpenmrsData {
+	
+	public static final long serialVersionUID = 1L;
+
+	private Integer conditionId;
+	
+	private CodedOrFreeText condition;
+
+	private ConditionClinicalStatus clinicalStatus;
+
+	private ConditionVerificationStatus verificationStatus;
+
+	private Long previousVersion;
```

Comment:
Are you sure that the previous version is a long?

Reply:
It is of type Condition. Modified

Reply:
I have modified this to be of type Condition. I have also added the type in the hibernate mapping and liquibase changeset as `type="org.openmrs.Condition"`